nixos/modules/services/misc/gitlab.nix

```diff
@@ -247,8 +247,7 @@ in {
       };
 
       databasePassword = mkOption {
-        type = types.str;
-        default = "";
+        type = types.path;
```

Comment:
aren't paths copied into the nix store as well? I stumbled upon this in #29298

Reply:
As far as I understand, neither the value of type str or path is copied directly to the store. The problem is that the secretsYml file *was*. Instead, that file (which is still copied to the store) is merely a template which we copy out substitute into at runtime from files whose path are specified here.

Reply:
Actually, I suppose the configuration is stored, but it doesn't matter, since the path, not the contents is what's made world-readable. At best you can now see *where* the secret is stored, but it should still be secure (if permissions are correct.)

Reply:
@srhb good work on making NixOS more secure! With regards to password options having type `path` see this comment by @domenkozar: https://github.com/NixOS/nixpkgs/issues/24288#issuecomment-344681430.